app/code/Magento/MediaGalleryUi/Model/SearchCriteria/CollectionProcessor/FilterProcessor/Directory.php

```diff
@@ -19,7 +19,16 @@ class Directory implements CustomFilterInterface
     public function apply(Filter $filter, AbstractDb $collection): bool
     {
         $value = $filter->getValue() !== null ? str_replace('%', '', $filter->getValue()) : '';
-        $collection->getSelect()->where('path REGEXP ? ', '^' . $value . '/[^\/]*$');
+
+        /**
+         * Use BINARY comparison for case-sensitive path filtering.
+         * Without BINARY, MySQL's default case-insensitive comparison would match
+         * directories like "Testing" and "testing" as the same, leading to incorrect
+         * file visibility across directories with different case variations.
+         * The regex '^{path}/[^\/]*$' ensures we only match files directly in the
+         * specified directory, not in subdirectories.
+         */
+        $collection->getSelect()->where('BINARY path REGEXP ? ', '^' . $value . '/[^\/]*$');
```

Comment:
Suggestion: Consider to add fallback behaviour with error handling:
```suggestion
       try {
    $collection->getSelect()->where('BINARY path REGEXP ? ', '^' . $value . '/[^\/]*$');
} catch (\Exception $e) {
    // Log error and fallback to non-BINARY comparison
    $this->logger->error('BINARY regex failed, falling back: ' . $e->getMessage());
    $collection->getSelect()->where('path REGEXP ? ', '^' . $value . '/[^\/]*$');
}
```

Reply:
Hello @engcom-Hotel,

The static test is giving warning on Magento copywriter comment. Do we have to change it to B2B copywriter comment?
```php
* Copyright 2019 Adobe
* All Rights Reserved.
 ```
 

> Suggestion: Consider to add fallback behaviour with error handling:

Great suggestion! I agree that adding error handling makes the solution more robust. I've added a few small modifications:
```php
try {
    $collection->getSelect()->where('BINARY path REGEXP ? ', '^' . $value . '/[^\/]*$');
} catch (\Exception $e) {
    // Log the error for debugging but continue with case-insensitive fallback
    // Note: This fallback means directory filtering will not be case-sensitive
    $this->logger->error('MediaGallery Directory Filter: BINARY REGEXP not supported, using case-insensitive fallback: ' . $e->getMessage());
    $collection->getSelect()->where('path REGEXP ? ', '^' . $value . '/[^\/]*$');
}
```
I was not sure how to edit the code suggestion to add my changes.

Reply:
> The static test is giving warning on Magento copywriter comment. Do we have to change it to B2B copywriter comment?

You need to refer the files which is mentioned in the failed test, for example, consider the following screenshot:

<img width="2014" height="516" alt="image" src="https://github.com/user-attachments/assets/41a98b29-d664-48df-a20a-acf79aba1018" />

The above screenshot is from the below static test failure:

https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/39502/fa70983fcf29df8efaba5329f6467951/Statics/report-sanity-ce.html

Reply:
> I was not sure how to edit the code suggestion to add my changes.

You can directly edit this file from your local in the same branch, and commit as usual.

Reply:
Thanks for the guidance. Let commit the changes.